gui/src/streamwindow.cpp

```diff
@@ -61,6 +72,21 @@ void StreamWindow::Init()
 
 	resize(connect_info.video_profile.width, connect_info.video_profile.height);
 	show();
+	
+	// Start in full screen mode.
+	// Fred - No! - Actually on RPi we don't want this window at all!
+	//ToggleFullscreen();
+	// Need a Quit hotkey as with the Pi the main framebuffer is used by the hw decoder
+	// and it seems hard to quit.
+	
+	auto rpi_quit_action = new QAction(tr("Quit"), this);
```

Comment:
Since the RPi hw decode+draw is unconventionally just pumping pixels ontop of everything I need a hotkey to Quit with.
This is not a good place to have it as if there's a problem somewhere (as Iv'e experienced) the screen stays but the keypress doesn't register.
I'd need a more app global context to receive this Quit hotkey press. 

Where could I add that instead?

(I also want to figure out some kind of 'minimize' or 'hide', but since this is a custom window I don't know yet how to do that so I'll look at that part later)

Reply:
I think the optimal solution would be to be able to place this custom window not always in fullscreen, but exactly at the place where the Qt window is shown, so in the end you wouldn't even notice it renders differently than with ffmpeg+OpenGL. But I am not sure if this is even possible with what is offered by the pi.

If it's not possible, we should probably always have a Qt window displayed in fullscreen "behind" the custom video display, so this fullscreen Qt window can catch all the key presses and we just kill the video output when this window is closed.